rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -183,11 +184,49 @@ public void setArtifacts(Link artifacts) {
         }
     }
 
-    public static RunExt create(WorkflowRun run) {
-        final RunExt runExt = new RunExt();
+    /** Computes timings after the stages have been set up
+     *  That means timing of the stage has been computed, and the stages are sorted
+     */
+    public static RunExt computeTimings(RunExt runExt) {
+
+        // Pause is the sum of stage pauses
+        for (StageNodeExt stage : runExt.getStages()) {
+            runExt.setDurationMillis(runExt.getPauseDurationMillis() + stage.getPauseDurationMillis());
+        }
+
+        if (!runExt.getStages().isEmpty()) {
+            // We're done when the last stage is, and it is the result of the overall run
+            FlowNodeExt lastStage = runExt.getLastStage();
+            runExt.setEndTimeMillis(lastStage.getStartTimeMillis()+lastStage.getDurationMillis());
+            lastStage.setStatus(runExt.getStatus());
+        }
+
+        long currentTimeMillis = System.currentTimeMillis();
```

Comment:
@kzantow Correct, it's just for in-progress executions.  Although the diff shows this as new, it is just refactored out from the previous createMethod (to allow for the new FlowAnalyzer to invoke it separately from the other creation routines).